### ISSUE TITLE
coder specified in the input PCollection is not a KvCoder

### ISSUE THREAD

**Issue by jto** [spotify/scio#4636]
Scio used to support using a KV transform with `applyTransform` for example the following was working:

```scala
val groupToBatches = GroupIntoBatches.ofSize[Int, FooBar](batchSize)
coll.map { v => KV.of(1, v) }.applyTransform("Group into batches", groupToBatches)
```

With Scio 0.11.13, the same code will now throw `coder specified in the input PCollection is not a KvCoder`.

Note: using the dedicated `applyKvTransform` method does not help.

**Comment by RustedBones** [comment]
This was only working for scio `0.11.11` and `0.11.12`. we had to revert the coder changes due to the binary incompatibility introduced by the macro changes.

This is working properly in scio `0.12.+`

I'm not sure if we want to backport this fix to the `0.11` branch though. @clairemcginty any take on that ?

**Comment by clairemcginty** [comment]
Hmm... I have a feeling we'll end up doing a final 0.11.x branch release in about a month, probably not a bad idea to include this?

**Comment by jto** [comment]
@RustedBones it worked for sure with older versions of Scio. The example I shared is from an old hack project using `0.10.0` .

**Comment by RustedBones** [comment]
Ok, the commit that triggered the regression is [this one](https://github.com/spotify/scio/commit/a8170ccd7ae6fce467f1220e2999add4094fc3bc).

Before that time, coders were unwrapped, which was actually considered as a [bug](https://github.com/spotify/scio/issues/3895#issuecomment-884445467).
Starting from `v0.11.1`, wrapping was again propagated, and failing beam equality check on `GroupIntoBatches`.

IMHO, this was working on scio `0.10.x` only by chance because wrapping was not done as expected.

On the wrapping topic, beam is getting stricter on coders, and checks strict equality. See [here](https://github.com/spotify/scio/pull/4549#issuecomment-1286900818). Maybe we should question the wrapping value vs maintainability.

**Comment by RustedBones** [comment]
Since 0.13 is out, we won't backport the fix to 0.11.
